homeassistant/components/lutron/binary_sensor.py

```diff
@@ -1,34 +1,42 @@
 """Support for Lutron Powr Savr occupancy sensors."""
 from __future__ import annotations
 
+from collections.abc import Mapping
+import logging
+from typing import Any
+
 from pylutron import OccupancyGroup
 
 from homeassistant.components.binary_sensor import (
     BinarySensorDeviceClass,
     BinarySensorEntity,
 )
+from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
-from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+from homeassistant.helpers.typing import DiscoveryInfoType
 
 from . import LUTRON_CONTROLLER, LUTRON_DEVICES, LutronDevice
 
+_LOGGER = logging.getLogger(__name__)
+
 
-def setup_platform(
+async def async_setup_entry(
     hass: HomeAssistant,
-    config: ConfigType,
-    add_entities: AddEntitiesCallback,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
-    """Set up the Lutron occupancy sensors."""
-    if discovery_info is None:
-        return
-    devs = []
-    for area_name, device in hass.data[LUTRON_DEVICES]["binary_sensor"]:
-        dev = LutronOccupancySensor(area_name, device, hass.data[LUTRON_CONTROLLER])
-        devs.append(dev)
+    """Set up the Lutron binary_sensor platform.
 
-    add_entities(devs)
+    Adds occupancy groups from the Main Repeater associated with the
+    config_entry as binary_sensor entities.
+    """
+    entities = []
+    for area_name, device in hass.data[LUTRON_DEVICES]["binary_sensor"]:
+        entity = LutronOccupancySensor(area_name, device, hass.data[LUTRON_CONTROLLER])
+        entities.append(entity)
+    async_add_entities(entities, True)
```

Comment:
```suggestion
    async_add_entities(entities)
```

Reply:
this is desired so that we get the current state after the entity is setup.

Reply:
Hmm, we did not do this before, I'll recheck this later